### DISCUSSION TITLE
January 2024 Update Report

### DISCUSSION THREAD

**Discussion by MotoLegacy** [nzp-team/nzportable#645] [Announcements]
### Introduction
This post is also available on [ModDB](https://www.moddb.com/games/nazi-zombies-portable/news/january-2024-update-report).

Sorry for the delay!
Real life has begun catching up again, so this month was a lot of me in maintenance mode, fixing bugs and finishing up some stuff on the backburner! That isn't to say the content this month wasn't interesting though :)

I've seen a lot of love on [Patreon](https://patreon.com/cypressimplex) recently and for that I want to say, thanks a ton! I'll continue to make it worth your while, even if things slow down a bit. I always put a lot of time and consideration into my contributions.

### Finalized Support for NZ:P Beta Maps

![image](https://github.com/nzp-team/nzportable/assets/26030250/614b0c7f-e13e-402f-a398-225a09dc8d73)

Hoorah! Compatibility with classic maps from the "Beta" (2011) era has been released on all platforms. I had done a lot of work to get this off the ground, testing dozens of maps individually, and creating + implementing a [new major build component](https://github.com/nzp-team/qchashtablegenerator) to the development pipeline in order to sufficiently and efficiently accommodate most maps. Please note that this is **not perfect**. There are maps that have unavoidable problems based on how they are built. Some have problems but can be mitigated, others do not. This is due to a lot of core/architectural game related differences, but I am very happy with the results regardless! 

So [grab one (or some!) of your favorite "Beta" maps](https://github.com/nzp-team/nzportable/discussions/158) and give them a try!

### Enhancements for Standard `func_door` Quake Entities

Mappers rejoice! `func_door` entities now have enhanced and fixed behaviors allowing them to be better integrated into your levels now, including support for opening Waypoints! No more `func_wall` `killtarget` hacks :)

### Weapon System Enhancements

https://github.com/nzp-team/nzportable/assets/26030250/4519d9d8-f3e7-405f-91b3-763dad100f65

I've done quite a bit of cleanup and some feature restoration for NZ:P's Weapon Core. Here are the big bullet points:

* Weapon Spread now increases over time when Hip Firing
* A lot of core internal systems are being migrated to primarily data-driven implementations, for custom weapon support in the future :)
* General Code Cleanup
* Sniper Sway is FINALLY implemented on FTEQW, because I forgot it was absent for years.
* Wonder Weapons no longer play the Pack-A-Punch fire sound effect
* Toggle ADS mode was re-written to be less broken

https://github.com/nzp-team/nzportable/assets/26030250/4f6b9b40-9d02-4ccc-b2a1-ce33724968a6

Sniper Sway is also completely absent if you have Deadshot Daiquiri, adding more usefulness to the Perk while improving Parity :)

### Co-Op Pause/Chat Indicators

https://github.com/nzp-team/nzportable/assets/26030250/3cc965dd-6fcf-4235-a829-5f7b37f2c550

Co-Op Players can now see when someone is Typing in Chat or has their game Paused, assisting in determining whether or not you need to provide assistance.

### FTE View-Model Rewrite

https://github.com/nzp-team/nzportable/assets/26030250/5f54f4f2-0ee0-4f0a-9b64-6a9c6225dc67

The way View Models are rendered and implemented in CSQC have been completely redone from scratch. Here's the gist of what this means for you:

* Fixes a long-standing interpolation bug when switching models
* Accurate walk and sprint bobbing compared to other platforms
* Unified weapon definitions for Aim Down Sight positions
* Removes all issues with playing at low frame-rates
* Viewmodel "slack" (has to catch up with fast camera movements)

I also took the opportunity to add the idle and subtle Aim bobbing to the other platforms too, so everyone benefited from this :)

### Screen Fading Capabilities

https://github.com/nzp-team/nzportable/assets/26030250/edb9b2d9-18d2-44af-855f-d48ef3b0f373

https://github.com/nzp-team/nzportable/assets/26030250/29627328-b1dc-4ec7-9ff0-81c49f9aa25a

(Mind the broken viewbob!)

While fixing a lot of the CSQC implementation's bloated protocols to improve networking performance, I decided to enhance the Game Over fade system to be more modular, allowing for QC-specified fade types, durations, and colors. This allowed for the game to fade in at start, and for a proper Nuke Power-Up flash!

It's also on all platforms now, woo!

### Double Tap Perk-A-Cola Fixes

![image](https://github.com/nzp-team/nzportable/assets/26030250/115384c4-bd63-4373-8a1c-361194253847)

This one was a little hard to visualize, so I offered my previous explanation in place. The summary: Double Tap now properly impacts automatic weapons in general, and has a more accurate-feeling stat :)

### Documentation Updates

Reminder that I maintain a [Documentation page](https://docs.nzp.gay)! I've updated the Trenchbroom template base to include the automatic dark mode detection, and added information on some of the NZ:P-exclusive trigger entities, including a new trigger, [`trigger_awardpoints`](https://docs.nzp.gay/mapping/trigger_awardpoints), which can be used to implement Perk-A-Cola change underneath the machine :)

### Barricade Re-model

https://github.com/nzp-team/nzportable/assets/26030250/a9cd6be4-7720-4b57-836d-7fa87014758e

Thanks to @MissLavender-LQ for contributing a new Barricade mesh, that is more proportionally accurate to Call of Duty, and has a snazzy new texture!

### IN-PROGRESS: Major UI Overhaul

https://github.com/nzp-team/nzportable/assets/26030250/773dae59-eb3f-4b2c-8a10-63766c5d230a

I have begun work on re-writing the CSQC menu system to test out a new UI design, that will be more user-personalized and interesting to look at! Menu backgrounds are taken from YOUR custom maps folders, and the menu music is configurable via a `cvar`, so you will be able to make NZ:P your own a little more than before.

![image](https://github.com/nzp-team/nzportable/assets/26030250/72fd4fcb-7080-4545-98a9-242d24ddc689)

See you next month!

**Thanks to my [Patrons](https://www.patreon.com/CypressImplex)!**

- Rinse a Gateau Mandate
- Benjamin Engelhardt
- mochi
- Ryan Baldwin
- Derped Crusader
- MissLavender
- LEXX
- Dawai
- Tony The Bird
- Jeiku
- Marcos Mata
- Shaun Zafar
- Swoop
- Bobthebob
- DukeofLeet

**Comment by PSRYT** [comment]
Wait, when installing the custom maps (beta), I have this data folder, what do i do with it?

**Reply by MotoLegacy** [reply]
Following the map install standard, you just drag the `data` directory in the `nzp` folder (i.e., `nzp/data/ankunft`)

**Reply by RatchetGAMES** [reply]
can you play custom maps in the browser port of nzp

**Reply by Peter0x44** [reply]
Yes. For things following the standard, rename .zip to .pk3 and drag it on the window.

**Comment by xxunknxwnxx** [comment]
Yes!! thank you so much for your hard work <3

**Comment by lavenderdotpet** [comment]
:3 i was @ :333

**Comment by JustinKyle11** [comment]
heya! would like to ask why multi doesnt work in browser. it always says giving up
like this
would love to see an reply, thanks!
![image](https://github.com/nzp-team/nzportable/assets/96884191/619f6c2f-3262-4667-b7b4-5323625524e5)

**Comment by DeevDaRabbit** [comment]
Would I be allowed to help with improving the sound design?

**Comment by Five-seveN1993** [comment]
How do i download for windows 11

**Reply by Peter0x44** [reply]
https://github.com/nzp-team/nzportable/releases/latest
You want nzportable-win64.zip from here, extract everything, and run nzportable64-sdl.exe

**Reply by Five-seveN1993** [reply]
Thanks!

**Reply by Five-seveN1993** [reply]
thank you!!!

**Comment by Five-seveN1993** [comment]
I started playing the game fixed some control stuff and the game doesn't use my mouse to look around anymore, please help

**Reply by Five-seveN1993** [reply]
i can use controller still though

**Reply by Peter0x44** [reply]
This is not the appropriate avenue to ask for support.
Pick any of the options mentioned here:
https://github.com/nzp-team/nzportable/discussions/550
